src/Spelling/WindowsSpellChecker.cs

```diff
@@ -0,0 +1,752 @@
+
+        private delegate NativeMethods.Ptec ProofCloseLex(IntPtr id, IntPtr lex, bool force);
+
+        private delegate NativeMethods.Ptec ProofInit(out IntPtr pid, ref NativeMethods.ProofParams pxpar);
+
+        private delegate NativeMethods.Ptec ProofOpenLex(
+            IntPtr id, ref NativeMethods.ProofLexIn plxin, ref NativeMethods.ProofLexOut plxout);
+
+        private delegate NativeMethods.Ptec ProofSetOptions(IntPtr id, uint iOptionSelect, uint iOptVal);
+
+        private delegate NativeMethods.Ptec ProofTerminate(IntPtr id, bool fForce);
+
+        private delegate NativeMethods.Ptec SpellerAddUdr(
+            IntPtr sid, IntPtr lex, [MarshalAs(UnmanagedType.LPTStr)] string add);
+
+        private delegate IntPtr SpellerBuiltInUdr(IntPtr sid, ProofLexType lxt);
+
+        private delegate NativeMethods.Ptec SpellerCheck(
+            IntPtr sid, SpellerCommand scmd, ref NativeMethods.Wsib psib, ref NativeMethods.Wsrb psrb);
+
+        private delegate NativeMethods.Ptec SpellerClearUdr(IntPtr sid, IntPtr lex);
+
+        private delegate NativeMethods.Ptec SpellerDelUdr(
+            IntPtr sid, IntPtr lex, [MarshalAs(UnmanagedType.LPTStr)] string delete);
+
+        internal enum ProofLexType : uint
+        {
+            ChangeAlways = 1,
+            ChangeOnce = 0,
+            Exclude = 3,
+            IgnoreAlways = 2,
+            Main = 4,
+            Max = 6,
+            SysUdr = 5,
+            User = 2
+        }
+
+        internal enum PtecMajor : uint
+        {
+            BufferTooSmall = 6,
+            IoErrorMainLex = 3,
+            IoErrorUserLex = 4,
+            ModuleError = 2,
+            ModuleNotLoaded = 8,
+            NoErrors = 0,
+            NotFound = 7,
+            NotSupported = 5,
+            OutOfMemory = 1
+        }
+
+        internal enum PtecMinor : uint
+        {
+            EntryTooLong = 0x8f,
+            FileCreate = 0x8a,
+            FileOpenError = 0x92,
+            FileRead = 0x88,
+            FileShare = 0x8b,
+            FileTooLargeError = 0x93,
+            FileWrite = 0x89,
+            InvalidCmd = 0x85,
+            InvalidEntry = 0x8e,
+            InvalidFormat = 0x86,
+            InvalidId = 0x81,
+            InvalidLanguage = 150,
+            InvalidMainLex = 0x83,
+            InvalidUserLex = 0x84,
+            InvalidWsc = 130,
+            MainLexCountExceeded = 0x90,
+            ModuleAlreadyBusy = 0x80,
+            ModuleNotTerminated = 140,
+            OperNotMatchedUserLex = 0x87,
+            ProtectModeOnly = 0x95,
+            UserLexCountExceeded = 0x91,
+            UserLexFull = 0x8d,
+            UserLexReadOnly = 0x94
+        }
+
+        [Flags]
+        internal enum SpellerStates : uint
+        {
+            IsContinued = 1,
+            IsEditedChange = 4,
+            NoStateInfo = 0,
+            StartsSentence = 2
+        }
+
+#pragma warning disable CA1717 // Only FlagsAttribute enums should have plural names
```

Comment:
💭 Seems like this is a bug in CA1717 if we're still seeing it reported

Reply:
No longer applies